### PR TITLE
Add gtk composite template support

### PR DESCRIPTION
**Pull request by badcel**
Remarks:
- HandlerNotFoundException: Moved into separate file
- SignalConnector: Moved into separate file (it is used by Builder and Widget) + change of constructor / method signature

### CODE REVIEW

**Comment by badcel** [comment]
Do you have any feedback for me? Are you interested in the code?

**Comment by cra0zy** [comment]
The change is of larger caliber so it will take me a bit of time to review.

I am interested in merging the code, I just don't have the time to properly review it right now.

File: Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.CompositeWidget.CSharp/Gtk_Composite_Widget.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using System;
+using Gtk;
+
+namespace GtkNamespace
+{
+    [Template("Gtk_Composite_Widget.glade")]
+    [TypeName(nameof(Gtk_Composite_Widget))]
+    public class Gtk_Composite_Widget : Bin
```
Comment (cra0zy) [inline Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.CompositeWidget.CSharp/Gtk_Composite_Widget.cs]: C# naming scheme please: `GtkCompositeWidget`

Comment (badcel) [reply Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.CompositeWidget.CSharp/Gtk_Composite_Widget.cs]: Done

Comment (cra0zy) [reply Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.CompositeWidget.CSharp/Gtk_Composite_Widget.cs]: You have missed several places where there is `Gtk_Composite_Widget`.

Comment (badcel) [reply Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.CompositeWidget.CSharp/Gtk_Composite_Widget.cs]: Okay, updated all occurrences

**Comment by knuxbbs** [comment]
@badcel needs some help with this pull request?

**Comment by badcel** [comment]
I assume that @harry-cpp ist waiting with this feature for a new major release? If the pr should be merged I can fix it.

**Comment by zii-dmg** [comment]
This code only works with first created `[Template]` widget because it not differentiates types:
```csharp
static bool IsGtkTemplate;
static Dictionary<FieldInfo, string> GtkTemplateFieldBindings = new Dictionary<FieldInfo, string>();
```
Save every other widget will have `IsGtkTemplate = true` and dictionary of foreign fields.

Solutions is to use type dictionary:
```csharp
static Dictionary<Type, FieldBindings> Templates; // FieldBindings = Dictionary<FieldInfo, string>
```
Type will be in dictionary if it is template and will have its own field bindings.

\
Also I want to say that without template support I can't use Glade designer for my custom widgets. Now Glade files is only working for `Window` objects. And without designer/previewer using any UI framework is not very productive.

**Comment by badcel** [comment]
I do not plan to pick this up again. If there is need feel free to continue the work.

**Pull request closed without merging**